internal/service/networkmanager/core_network_policy_document_data_source.go

```diff
@@ -160,8 +160,9 @@ func DataSourceCoreNetworkPolicyDocument() *schema.Resource {
 										ValidateFunc: verify.ValidRegionName,
 									},
 									"asn": {
-										Type:     schema.TypeInt,
-										Optional: true,
+										Type:         schema.TypeInt,
```

Comment:
This should be changed to a `schema.TypeString`, which will be handled internally as an `int64`. We can take advantage of the Terraform parser which will allow some string values, such as numbers and boolean values to omit the quotes